CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+    rules described in `Running static code analysis locally <#running-static-code-analysis-locally>`__. section.
+
+-   Read this excellent `article <http://chris.beams.io/posts/git-commit/>`__. on
+    commit messages and adhere to them. It makes the lives of those who come
+    after you a lot easier.
+
+
+
+Development Environments
+========================
+
+There are two environments, available on Linux and macOS, that you can use to
+develop Apache Airflow:
+
+-   `Local virtualenv development environment <#local-virtualenv-development-environment>`_
+    that supports running unit tests and can be used in your IDE.
+
+-   `Breeze Docker-based development environment <#breeze-test-development-environment>`_
+
+The table below summarizes differences between the two environments:
+
+
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+| **Development Environment** | **Test Coverage**                                                  | **Setup**                      |
++=============================+====================================================================+================================+
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+| Local virtualenv            | - (-) limited to unit tests with no dependencies                   | - (-) manual setup             |
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+| Breeze environment          | - (+) supports integration and unit tests                          | - (+) automated setup          |
+|                             | - (+) reproducible in other dev environments and Travis CI builds  | - (-) consumes docker space    |
+|                             |                                                                    | - (-) high CPU utilization     |
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+
+Typically, you are recommended to use both of these environments depending on
+your needs.
+
+Local virtualenv Development Environment
+----------------------------------------
+
+All details about using and running local virtualenv enviroment for Airflow can
+be found
+
+in `LOCAL_VIRTUALENV.rst <LOCAL_VIRTUALENV.rst>`__.
+
+Benefits:
+
+-   Packages are installed locally. No container environment is required.
+
+-   You can benefit from local debugging within your IDE.
+
+-   You can retain access to a Python virtualenv in your local IDE that meets
+    necessary requirements and supports autocompletion and running tests
```

Comment:
Elena: what requirements?
Jarek: i think this whole sentence can be replaced with something along "you can use it in your IDE for autocompletion and running tests directly from the IDE" or similar. Most developers will know how to do it (if they use such IDEs) and we have some examples in the LOCAL_VIRTUALENV.rst and BREEZE.rst how to set it up.)